src/App.js

```diff
@@ -35,21 +37,76 @@ const styles = theme => ({
 });
 
 class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      authContext: {
+        authData: {},
+        isAuthenticated: this.isAuthenticated,
+        setCurrentUser: this.setCurrentUser,
+        setCurrentUserProfile: this.setCurrentUserProfile
+      }
+    };
+  }
+
+  componentDidMount = () => {
+    // Sync auth session stored in the localStorage with Context
+    // TBH, I'm not sure if we even need context
+    this.setCurrentUser(AuthService.getSessionAuthData());
+  };
+
+  isAuthenticated = () => {
+    const {
+      userId,
+      expiresAt,
+      auth0AccessToken,
+      graphcoolToken
+    } = this.state.authContext.authData;
+    return (
+      Date.now() < expiresAt && userId && auth0AccessToken && graphcoolToken
+    );
+  };
+
+  setCurrentUser = authData => {
```

Comment:
These methods are confusingly named. Should this be called `setAuthData` or `setAuthMetadata` instead?